xwiki-pro-macros-api/src/main/java/com/xwiki/macros/excerptinclude/internal/macro/ExcerptIncludeMacro.java

```diff
@@ -0,0 +1,145 @@
+import org.xwiki.display.internal.DocumentDisplayer;
+import org.xwiki.display.internal.DocumentDisplayerParameters;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.rendering.block.Block;
+import org.xwiki.rendering.block.MacroBlock;
+import org.xwiki.rendering.block.TableBlock;
+import org.xwiki.rendering.block.TableCellBlock;
+import org.xwiki.rendering.block.TableHeadCellBlock;
+import org.xwiki.rendering.block.TableRowBlock;
+import org.xwiki.rendering.block.WordBlock;
+import org.xwiki.rendering.block.XDOM;
+import org.xwiki.rendering.block.match.ClassBlockMatcher;
+import org.xwiki.rendering.macro.MacroExecutionException;
+import org.xwiki.rendering.transformation.MacroTransformationContext;
+import org.xwiki.security.authorization.ContextualAuthorizationManager;
+import org.xwiki.security.authorization.Right;
+import org.xwiki.stability.Unstable;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xwiki.macros.AbstractProMacro;
+import com.xwiki.macros.excerptinclude.macro.ExcerptIncludeMacroParameters;
+
+/**
+ * Implements the 'excerpt-include' macro functionality to include excerpts from other documents.
+ * This macro serves as a bridge for the Confluence Excerpt Include macro.
+ *
+ * @version $Id$
+ * @since 1.14.4
+ */
+@Component
+@Named("excerpt-include")
+@Singleton
+@Unstable
+public class ExcerptIncludeMacro extends AbstractProMacro<ExcerptIncludeMacroParameters>
+{
+    @Inject
+    private ContextualAuthorizationManager contextualAuthorization;
+
+    @Inject
+    @Named("content")
+    private DocumentDisplayer contentDisplayer;
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    /**
+     * Constructs an instance of ExcerptIncludeMacro. Initializes the macro descriptor with a specific name,
+     * description, and parameter class type.
+     */
+    public ExcerptIncludeMacro()
+    {
+        super("Excerpt include",
+            "Bridge for the Confluence Excerpt Include macro, includes excerpts from other document.",
+            ExcerptIncludeMacroParameters.class);
+    }
+
+    @Override
+    public boolean supportsInlineMode()
+    {
+        return false;
+    }
+
+    @Override
+    protected List<Block> internalExecute(ExcerptIncludeMacroParameters parameters, String content,
+        MacroTransformationContext context) throws MacroExecutionException
+    {
+        DocumentReference reference = parameters.getReference();
+        XWikiContext xcontext = contextProvider.get();
+        XWikiDocument document;
+        try {
+            document = xcontext.getWiki().getDocument(reference, xcontext);
+        } catch (XWikiException e) {
+            throw new MacroExecutionException(String.format("Failed to get document for reference [%s]", reference), e);
+        }
+
+        if (!this.contextualAuthorization.hasAccess(Right.VIEW, reference)) {
```

Comment:
Should we also check for license? (e.g. https://github.com/xwikisas/application-confluence-migrator-pro/blob/a0ace06777aa4465144faa86e26f038c8fe75bce/application-confluence-migrator-pro-default/src/main/java/com/xwiki/confluencepro/internal/ConfluenceFilteringListener.java#L111 )

Reply:
It is not necessary because the `internalExecute()` method is called in the `execute()` method from AbstractProMacro, which check for licensure https://github.com/xwikisas/xwiki-pro-macros/blob/main/xwiki-pro-macros-api/src/main/java/com/xwiki/macros/AbstractProMacro.java#L115. 
I hope I'm not wrong.

Reply:
Indeed, didn't noticed it